src/express-server.ts

```diff
@@ -0,0 +1,229 @@
+  AUTH_REQUIRED: "Authorization header is required",
+  AUTH_EMPTY: "Authorization header cannot be empty",
+  SESSION_REQUIRED: "mcp-session-id header is required",
+  TRANSPORT_NOT_FOUND: "Transport not found",
+  INVALID_SESSION: "Invalid session ID or method",
+  INTERNAL_ERROR: "Internal server error",
+} as const;
+
+class HTTPError extends Error {
+  status: number;
+
+  constructor(status: number, message: string) {
+    super(message);
+    this.status = status;
+    this.name = "HTTPError";
+  }
+}
+
+// Extended Express Request interface
+declare global {
+  namespace Express {
+    interface Request {
+      ctx: Props;
+      mcpServer: McpServer;
+    }
+  }
+}
+
+// Transport storage
+type TransportMap = Map<string, StreamableHTTPServerTransport>;
+
+const httpTransportMap: TransportMap = new Map();
+
+/**
+ * Type guard to check if request body contains an initialize request
+ */
+function isInitializeReq(body: unknown): body is InitializeRequest {
+  const isInitial = (data: unknown): boolean =>
+    InitializeRequestSchema.safeParse(data).success;
+
+  if (Array.isArray(body)) return body.some(isInitial);
+  return isInitial(body);
+}
+
+/**
+ * Validates authorization header and returns it if valid
+ */
+function validateAuthHeader(authHeader: unknown): string {
+  if (typeof authHeader !== "string") {
+    console.error("Authorization header is missing or not a string");
+    throw new HTTPError(401, ERROR_MESSAGES.AUTH_REQUIRED);
+  }
+
+  if (authHeader.length === 0) {
+    console.error("Invalid Authorization header: empty string");
+    throw new HTTPError(401, ERROR_MESSAGES.AUTH_EMPTY);
+  }
+
+  return authHeader;
+}
+
+/**
+ * Creates a new MCP server instance with the provided API key
+ */
+function createMCPServerInstance(apiKey: string): McpServer {
+  return createMCPServer({
+    logger: createConsoleLogger(LOG_LEVEL),
+    getSDK: () =>
+      new FireHydrantCore({
+        security: {
+          api_key: apiKey,
+        },
+      }),
+  });
+}
+
+/**
+ * Validates session ID header and returns it if valid
+ */
+function validateSessionId(sessionId: unknown): string {
+  if (typeof sessionId !== "string") {
+    throw new HTTPError(400, ERROR_MESSAGES.SESSION_REQUIRED);
+  }
+  return sessionId;
+}
+
+/**
+ * Gets transport for session ID or throws error if not found
+ */
+function getTransportForSession(
+  sessionId: string,
+): StreamableHTTPServerTransport {
+  const httpTransport = httpTransportMap.get(sessionId);
+  if (!httpTransport) {
+    console.error(`No transport found for sessionId: ${sessionId}`);
+    throw new HTTPError(400, ERROR_MESSAGES.TRANSPORT_NOT_FOUND);
+  }
+  return httpTransport;
+}
+
+/**
+ * Creates a new transport and connects it to the MCP server
+ */
+async function createAndConnectTransport(
+  mcpServer: McpServer,
+  req: Request,
+  res: Response,
+): Promise<void> {
+  const httpTransport = new StreamableHTTPServerTransport({
+    sessionIdGenerator: () => randomUUID(),
```

Comment:
if this basically an auth token - v4 uuid should not be used - generate random bytes